sos/report/plugins/aap_containerized.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2025 Nagoor Shaik <nshaik@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import os
+import subprocess
+from sos.report.plugins import Plugin, RedHatPlugin, PluginOpt
+
+
+class AAPContainerizedPlugin(Plugin, RedHatPlugin):
+    """Collects details about AAP Containerized setup
+    under a user's home directory"""
+
+    short_desc = "AAP Containerized Setup Plugin"
+    plugin_name = "aap_containerized"
+    profiles = ("sysmgmt", "ansible",)
+    packages = ("podman",)
+
+    option_list = [
+        PluginOpt(
+            "username",
+            default="",
+            val_type=[int, str],
+            desc="Provide username that was used to setup "
+            "AAP containerized installation"
+        )
+    ]
+
+    def setup(self):
+        username = self.get_option("username")
+        if not username:
+            self._log_debug("Username is mandatory to collect "
+                            "AAP containerized setup logs")
+            return
+
+        # Grab aap installation directory under user's home
+        user_home_directory = os.path.expanduser(f"~{username}")
+        aap_directory_name = self.path_join(user_home_directory, "aap")
+
+        # Don't collect cert and key files from the installation directory
+        if self.path_exists(aap_directory_name):
+            forbidden_paths = [
+                self.path_join(aap_directory_name, path)
+                for path in [
+                    "containers",
+                    "tls",
+                    "controller/etc/*.cert",
+                    "controller/etc/*.key",
+                    "eda/etc/*.cert",
+                    "eda/etc/*.key",
+                    "gateway/etc/*.cert",
+                    "gateway/etc/*.key",
+                    "hub/etc/*.cert",
+                    "hub/etc/*.key",
+                    "hub/etc/keys/*.pem",
+                    "postgresql/*.crt",
+                    "postgresql/*.key",
+                    "receptor/etc/*.crt",
+                    "receptor/etc/*.key",
+                    "receptor/etc/*.pem",
+                    "redis/*.crt",
+                    "redis/*.key",
+                ]
+            ]
+            self.add_forbidden_path(forbidden_paths)
+            self.add_copy_spec(aap_directory_name)
+        else:
+            self._log_debug(f"Directory {aap_directory_name} does not exist")
+
+        # Gather output of following podman commands as user
+        podman_commands = [
+            (f"sudo -u {username} -i podman info --debug", "podman_info"),
```

Comment:
We have a 'runas=user' option now for add_cmd_output() that may be better than calling sudo here. 
Check the sunbeam plugin for examples of how to use it, so we don't have to call 'sudo' here.

Reply:
@jcastill thanks for the review, I initially tried using the `runas` argument. However, its failing with permission denied errors as below. So, I had to rely on the literal commands, I can switch to `su` instead of `sudo` as used in the sunbeam plugin https://github.com/sosreport/sos/blob/main/sos/report/plugins/sunbeam.py#L84

~~~
[root@host tmp]# cat sosreport-host-2025-01-02-zovejyg/sos_commands/aap_containerized/podman_ps_all_json
time="2025-01-02T13:57:37+05:30" level=error msg="mkdir /run/user/0/libpod: permission denied"

[root@host tmp]# cat sosreport-host-2025-01-02-zovejyg/sos_commands/aap_containerized/podman_info
time="2025-01-02T13:57:36+05:30" level=error msg="mkdir /run/user/0/libpod: permission denied"
~~~

Reply:
@snagoor I got a very similar issue while working on the instructlab plugin, and had to drop runas there as well until I can investigate what's going on (which I'll do next week). 
I think with this issue you could continue with sudo until we find the problem with runas, then move on to using it in the future.